front/src/modules/towedRollingStock/hooks/useFilterTowedRollingStock.ts

```diff
@@ -0,0 +1,93 @@
+import { useEffect, useState } from 'react';
+
+import { osrdEditoastApi, type TowedRollingStock } from 'common/api/osrdEditoastApi';
+import { setFailure } from 'reducers/main';
+import { useAppDispatch } from 'store';
+import { castErrorToFailure } from 'utils/error';
+
+export interface TowedRollingStockFilters {
+  id?: number;
+  text: string;
+}
+
+const initialFilters: TowedRollingStockFilters = {
+  text: '',
+};
+
+function filterTowedRollingStocks(
+  towedRollingStockList: TowedRollingStock[],
+  filters: TowedRollingStockFilters
+) {
+  if (filters === initialFilters) return towedRollingStockList;
+  return towedRollingStockList.filter(({ id, name }) => {
+    if (filters.id !== undefined) {
+      return id === filters.id;
+    }
+    if (!filters.text) {
+      return true;
+    }
+    return name.trim().toLowerCase().includes(filters.text.trim().toLowerCase());
+  });
+}
+
+export default function useFilterTowedRollingStock() {
```

Comment:
For custom hooks, we use `const`

```suggestion
const useFilterTowedRollingStock = () => {
```